### PR TITLE
Add more bad-case testing of Resumption Tickets

### PR DESCRIPTION
**Pull request by anrossi**
Adds test cases which stress bounds-checking on resumption ticket parsing.
This PR largely addresses #1219

### CODE REVIEW

File: src/core/crypto.c [hunk 1]
```diff
@@ -2365,18 +2391,32 @@ QuicCryptoDecodeClientTicket(
             "[conn][%p] ERROR, %s.",
             Connection,
             "Client Ticket version unsupported");
-        Status = QUIC_STATUS_INVALID_PARAMETER;
+        goto Error;
+    }
+    if (ClientTicketLength - Offset < sizeof(uint32_t)) {
```
Comment (anrossi) [inline src/core/crypto.c:2396]: Update this to use addition instead of subtraction.

File: src/core/unittest/TicketTest.cpp [hunk 2]
```diff
@@ -93,6 +93,336 @@ TEST(ResumptionTicketTest, ClientEncDec)
...
+                7 + (uint16_t)(EncodedTPLength - CxPlatTlsTPHeaderSize) + (uint16_t)sizeof(ServerTicket),
+                InputTicketBuffer,
+                &DecodedTP,
+                &DecodedServerTicket,
+                &DecodedServerTicketLength,
+                &DecodedQuicVersion));
+    }
+
+    // Client TP length longer than actual
+     InputTicketBuffer[5] = (uint8_t)(TransportParametersLength - (uint8_t)CxPlatTlsTPHeaderSize) + 1;
```
Comment (nibanks) [inline src/core/unittest/TicketTest.cpp:335]:
```suggestion
    InputTicketBuffer[5] = (uint8_t)(TransportParametersLength - (uint8_t)CxPlatTlsTPHeaderSize) + 1;
```

**Comment by nibanks** [comment]
Looks good! Just need to fix up some Linux build issues and then I think it can be merged.

File: src/core/unittest/TicketTest.cpp [hunk 3]
```diff
@@ -208,6 +540,457 @@ TEST(ResumptionTicketTest, ServerEncDecNoAppData)
...
+                AlpnList,
+                sizeof(AlpnList),
+                &DecodedTP,
+                &DecodedAppData,
+                &DecodedAppDataLength));
+    }
+    InputTicketBuffer[5] = (uint8_t)sizeof(Alpn);
+
+    // Handshake TP length shorter than actual
+    for (uint8_t s = 0; s < EncodedTPLength - CxPlatTlsTPHeaderSize; ++s) {
```
Comment (nibanks) [inline src/core/unittest/TicketTest.cpp:877]:
```suggestion
    for (uint32_t s = 0; s < EncodedTPLength - CxPlatTlsTPHeaderSize; ++s) {
```
Fix this Code scanning failure please?

**Review by nibanks — APPROVED** [review]

**Pull request merged**